src/model/recoverers/BatchRecoverer.ts

```diff
@@ -0,0 +1,31 @@
+import { Recoverer } from './Recoverer'
```

Comment:
I would keep in mind this logic to see if would be suitable to put it inside the recoverer